### PR TITLE
Avoid memory copy when parsing log

### LINKED ISSUE #199: Avoid useless memory copy when parsing log entry.
### Description

Avoid useless memory copy when parsing log entry.

### PR DESCRIPTION
**Pull request by JackyWoo**
### Which issues of this PR fixes:

This PR try to fix #199 

### Change log:

- Avoid memory copy when parsing log

### CODE REVIEW

File: src/Service/NuRaftLogSegment.cpp [hunk 1]
```diff
@@ -572,7 +572,8 @@ int NuRaftLogSegment::loadLogEntry(int fd, off_t offset, LogEntryHeader * head,
         return -1;
     }
 
-    entry = LogEntryBody::parse(entry_str, head->term, head->data_length);
+    entry = LogEntryBody::parse(entry_str, head->data_length);
+    entry->set_term(head->term);
 
     delete[] entry_str;
```
Comment (lzydmxy) [inline src/Service/NuRaftLogSegment.cpp:578]: why not use buffer for entry_str instead of char pointer to avoid copy and delete.

Comment (JackyWoo) [reply src/Service/NuRaftLogSegment.cpp:578]: Because when we serialize data there is no header of buffer and this copy can not be avoided

Comment (lzydmxy) [reply src/Service/NuRaftLogSegment.cpp:578]: Got it, maybe better to use `buffer` to avoid manual memory release.

Comment (JackyWoo) [reply src/Service/NuRaftLogSegment.cpp:578]: No need to worry about memory deallocation issues for the method is only invoked synchronously.

Comment (JackyWoo) [reply src/Service/NuRaftLogSegment.cpp:578]: But I think we should not use raw pointer in `NuRaftLogSegment.cpp`, maybe fix in another PR.

Comment (lzydmxy) [reply src/Service/NuRaftLogSegment.cpp:578]: okay

**Comment by lzydmxy** [comment]
LGTM

**Pull request merged**